### PR TITLE
Add support for SQS queue tags

### PR DESCRIPTION
**Pull request by ewbankkit**
Fixes https://github.com/terraform-providers/terraform-provider-aws/issues/1983.

Acceptance tests:
```
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSQSQueue_tags'==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -run=TestAccAWSSQSQueue_tags -timeout 120m
=== RUN   TestAccAWSSQSQueue_tags
--- PASS: TestAccAWSSQSQueue_tags (39.54s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	39.549s
```
Ensure no regression:
```
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSQSQueue_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -run=TestAccAWSSQSQueue_basic -timeout 120m
=== RUN   TestAccAWSSQSQueue_basic
--- PASS: TestAccAWSSQSQueue_basic (38.16s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	38.170s
```

### CODE REVIEW

File: aws/resource_aws_sqs_queue.go [hunk 1]
```diff
@@ -16,19 +16,19 @@ import (
...
-	"receive_wait_time_seconds":         "ReceiveMessageWaitTimeSeconds",
-	"visibility_timeout_seconds":        "VisibilityTimeout",
-	"policy":                            "Policy",
-	"redrive_policy":                    "RedrivePolicy",
-	"arn":                               "QueueArn",
-	"fifo_queue":                        "FifoQueue",
-	"content_based_deduplication":       "ContentBasedDeduplication",
-	"kms_master_key_id":                 "KmsMasterKeyId",
-	"kms_data_key_reuse_period_seconds": "KmsDataKeyReusePeriodSeconds",
+var sqsQueueAttributeMap = map[string]string{
```
Comment (radeksimko) [inline aws/resource_aws_sqs_queue.go]: I appreciate this cleanup 👍

**Comment by AntonBazhal** [comment]
@ewbankkit @radeksimko This introduced breaking change to the plugin, since it requires "sqs:ListQueueTags" permission to be added to IAM policy.

**Comment by ewbankkit** [comment]
@AntonBazhal Yes, that was something I had thought about and we hit a similar thing with KMS tags - https://github.com/hashicorp/terraform/pull/12243. I'm still not sure there's an "official" way of dealing with these.

**Comment by AntonBazhal** [comment]
@ewbankkit The main problem is that this was released with a minor version bump and I spent half a day today trying to understand why all our builds started to fail, even though everything was OK yesterday.

**Pull request merged**